### PR TITLE
SUP-17625 Fix table plugin style configuration

### PR DESCRIPTION
**Pull request by yrucrem**
https://jira-gentics.atlassian.net/browse/SUP-17625

### CODE REVIEW

File: src/plugins/common/table/lib/table-plugin.js [hunk 1]
```diff
@@ -853,6 +855,83 @@ define([
...
+								options.push({
+									id: option.name,
+									icon: option.icon,
+									label: option.label
+								});
+							}
+						}
+					}
+
+					return {
```
Comment (deckdom) [inline src/plugins/common/table/lib/table-plugin.js:918]: `if (options.length === 0) return null`

File: src/plugins/common/table/lib/table-plugin.js [hunk 2]
```diff
@@ -853,6 +855,83 @@ define([
 				}
 			}
 		});
+
+		const haveCellConfig = Array.isArray(that.settings.cellConfig) && that.settings.cellConfig.length > 0;
```
Comment (deckdom) [inline src/plugins/common/table/lib/table-plugin.js:859]: Move into context function.
Return `null` when no options available.

File: src/plugins/common/table/lib/table-plugin.js [hunk 3]
```diff
@@ -853,6 +855,83 @@ define([
...
+				contextType: 'dropdown',
+				tooltip: i18n.t("button.cellstyle.tooltip"),
+				icon: Icons.TABLE_STYLE_CELLS,
+				context: function () {
+					const options = [];
+					const selType = TablePlugin.activeTable.selection.selectionType;
+					const usedClasses = new Set();
+
+					if (haveCellConfig) {
+						for (let i = 0; i < that.settings.cellConfig.length; i++) {
```
Comment (deckdom) [inline src/plugins/common/table/lib/table-plugin.js:874]: In function auslagern: `function xyx(config, currentOptions)`
```js
const currentOptions = {};

function xyz(config, currentOptions) {
  if (currentOptions[config[0].id]) {
    return;
  }
}

Object.values(xyz([], currentOptions);
```

File: doc/guides/source/plugin_table.textile [hunk 4]
```diff
@@ -117,16 +117,16 @@ Aloha.settings.plugins: {
...
 		rowConfig : [
-				{ name: 'table-style-bigbold',  iconClass: 'aloha-button-row-bigbold' },
-				{ name: 'table-style-redwhite', iconClass: 'aloha-button-row-redwhite' }
+				{ name: 'table-style-align-left',  icon: 'format_align_left', label: 'Align left' },
+				{ name: 'table-style-align-right', icon: 'format_align_right', label: 'Align right' }
 		],
 		cellConfig : [
-				{ name: 'table-style-bigbold',  iconClass: 'aloha-button-row-bigbold' },
-				{ name: 'table-style-redwhite', iconClass: 'aloha-button-row-redwhite' }
+				{ name: 'table-style-align-left',  icon: 'format_align_left', label: 'Align left' },
```
Comment (deckdom) [inline doc/guides/source/plugin_table.textile:128]: Add example with `iconHollow: true`

File: src/plugins/common/table/lib/table-plugin.js [hunk 5]
```diff
@@ -853,6 +855,83 @@ define([
...
+							const option = that.settings.cellConfig[i];
+
+							if (option.name && option.label) {
+								usedClasses.add(option.name);
+								options.push({
+									id: option.name,
+									icon: option.icon,
```
Comment (deckdom) [inline src/plugins/common/table/lib/table-plugin.js:881]: Add `iconHollow` mapping (optional property)

File: src/plugins/common/table/lib/table.js [hunk 6]
```diff
@@ -1667,6 +1667,7 @@ define([
 
 		this.selection.notifyCellsSelected();
 		this._removeCursorSelection();
+		this.tablePlugin.activeTable.selection.selectionType = 'column';
```
Comment (deckdom) [inline src/plugins/common/table/lib/table.js:1670]: Move before notifiy functions

File: doc/guides/source/plugin_table.textile (hunk 4 above)
Comment (yrucrem) [reply doc/guides/source/plugin_table.textile:128]: Done

File: src/plugins/common/table/lib/table-plugin.js (hunk 1 above)
Comment (yrucrem) [reply src/plugins/common/table/lib/table-plugin.js]: Done

File: src/plugins/common/table/lib/table-plugin.js (hunk 3 above)
Comment (yrucrem) [reply src/plugins/common/table/lib/table-plugin.js]: Done

File: src/plugins/common/table/lib/table-plugin.js (hunk 5 above)
Comment (yrucrem) [reply src/plugins/common/table/lib/table-plugin.js]: Done (also added in documentation).

File: src/plugins/common/table/lib/table-plugin.js (hunk 2 above)
Comment (yrucrem) [reply src/plugins/common/table/lib/table-plugin.js]: Done

File: src/plugins/common/table/lib/table.js (hunk 6 above)
Comment (yrucrem) [reply src/plugins/common/table/lib/table.js:1670]: Done.

**Pull request merged**